omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ModuleResource1_8.java

```diff
@@ -103,4 +128,43 @@ public static String getDisplay(Module instance) {
 		return instance.getName();
 	}
 	
+	@Override
+	public Object upload(MultipartFile file, RequestContext context) throws ResponseException, IOException {
+		moduleFactoryWrapper.checkPrivilege();
+		
+		File moduleFile = null;
+		Module module = null;
+		
+		try {
+			if (file == null || file.isEmpty()) {
+				throw new IllegalArgumentException("Uploaded OMOD file cannot be empty");
+			} else {
+				String filename = WebUtil.stripFilename(file.getOriginalFilename());
```

Comment:
E.g I don't see where in the legacyui they explicitly deal with dependent module like you are doing here, isn't that taken care of by the API?